### PR TITLE
Added validation block to prevent node_conf from setting Sockets

### PR DESCRIPTION
**Pull request by abbas1902**
Added a validation block to prevent Sockets from being set as part of node_conf
Manually tested using edited version of an existing slurm v6 blueprint.

Before (when running `terraform validate`):
```
Success! The configuration is valid.
```

Now:
```
Error: exit status 1

Error: Invalid value for variable

  on main.tf line 48, in module "debug_nodeset":
  48:   node_conf = {
  49:     Sockets = 2
  50:   }
    ├────────────────
    │ var.node_conf is map of number with 1 element

`Sockets` field is in conflict with `SocketsPerBoard` which is automatically
generated by SlurmGCP.
Instead, you can override the following fields: `Boards`, `SocketsPerBoard`, `CoresPerSocket`, and
`ThreadsPerCore`.
See: https://slurm.schedmd.com/slurm.conf.html#OPT_Boards and
https://slurm.schedmd.com/slurm.conf.html#OPT_Sockets_1"

This was checked by the validation rule at
modules/embedded/community/modules/compute/schedmd-slurm-gcp-v6-nodeset/variables.tf:32,3-13.
```

### Submission Checklist

Please take the following actions before submitting this pull request.

* Fork your PR branch from the Toolkit "develop" branch (not main)
* Test all changes with pre-commit in a local branch [#](https://goo.gle/hpc-toolkit#development)
* Confirm that "make tests" passes all tests
* Add or modify unit tests to cover code changes
* Ensure that unit test coverage remains above 80%
* Update all applicable documentation
* Follow Cluster Toolkit Contribution guidelines [#](https://goo.gle/hpc-toolkit-contributing)

### CODE REVIEW

File: community/modules/compute/schedmd-slurm-gcp-v6-nodeset/variables.tf [hunk 1]
```diff
@@ -29,6 +29,10 @@ variable "node_conf" {
   description = "Map of Slurm node line configuration."
   type        = map(any)
   default     = {}
+  validation {
+    condition     = lookup(var.node_conf, "Sockets", null) == null
+    error_message = "DEPRECATED: With the use of Boards in V6 the Sockets field is deprecated, use SocketsPerBoard instead."
```
Comment (mr0re1) [inline community/modules/compute/schedmd-slurm-gcp-v6-nodeset/variables.tf:34]: I would avoid words `DEPRECATED` and `V6`. Let's word it somewhat likes this: 
```
`Sockets` settings conflicts with `SocketsPerBoard` that is automatically computed by SlurmGCP.
Instead you can override `Boards`, `SocketsPerBoard`, `CoresPerSocket`, and `ThreadsPerCore`.
See: https://slurm.schedmd.com/slurm.conf.html#OPT_Boards
```

Comment (abbas1902) [reply community/modules/compute/schedmd-slurm-gcp-v6-nodeset/variables.tf]: That message seems a lot more descriptive, only change I made was to add slurm link about sockets for additional context on sockets and socketsperboard being mutually exclusive params, etc.

**Review by mr0re1 — APPROVED** [review]

File: community/modules/compute/schedmd-slurm-gcp-v6-nodeset/variables.tf [hunk 2]
```diff
@@ -29,6 +29,14 @@ variable "node_conf" {
   description = "Map of Slurm node line configuration."
   type        = map(any)
   default     = {}
+  validation {
+    condition     = lookup(var.node_conf, "Sockets", null) == null
+    error_message = <<-EOD
+    `Sockets` settings conflicts with `SocketsPerBoard` that is automatically computed by SlurmGCP.
```
Comment (mr0re1) [inline community/modules/compute/schedmd-slurm-gcp-v6-nodeset/variables.tf:35]: `s/settings/setting/`

Comment (abbas1902) [reply community/modules/compute/schedmd-slurm-gcp-v6-nodeset/variables.tf]: I believe this is addressed in the updated version but if not feel free to open this comment again

**Pull request merged**